src/Acuminator/Acuminator.Analyzers/StaticAnalysis/LongOperationStart/StartLongOperationWalker.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System;
+
+namespace Acuminator.Analyzers.StaticAnalysis.LongOperationStart
+{
+    public class StartLongOperationWalker : NestedInvocationWalker
+    {
+        private readonly PXContext _pxContext;
+        private readonly Action<Diagnostic> _reportDiagnostic;
+        private readonly DiagnosticDescriptor _descriptor;
+
+        public StartLongOperationWalker(SymbolAnalysisContext context, PXContext pxContext, DiagnosticDescriptor descriptor)
+            : base(context.Compilation, context.CancellationToken)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
```

Comment:
It is an overkill to check a cancellation token right in the constructor - it is checked immediately before the walker's creation.